components/readwise/sources/new-documents/new-documents.mjs

```diff
@@ -0,0 +1,64 @@
+import readwise from "../../readwise.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+export default {
+  key: "readwise-new-documents",
+  name: "New Documents",
+  description: "Emit new document",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    readwise,
+    timer: {
+      label: "Polling interval",
+      description: "Pipedream polls the Readwise API on this schedule",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+    location: {
+      type: "string",
+      label: "Location",
+      description: "(Optional) The document's location, can be one of: new, later, shortlist, archive, feed",
+      optional: true,
+    },
+    category: {
+      type: "string",
+      label: "Category",
+      description: "(Optional) The document's category, can be one of: article, email, rss, highlight, note, pdf, epub, tweet, video",
+      optional: true,
+    },
+  },
+  hooks: {
+    async activate() {
+      await this.processDocuments();
+    },
+  },
+  methods: {
+    async processDocuments(params) {
+      const { results: events } = await this.readwise.listDocuments({
+        params,
+      });
+      for (const event of events) {
+        this.emitEvent(event);
+      }
```

Comment:
The [document](https://pipedream.com/docs/components/api/#dedupe-strategies) says it stores 100 emitted events, and Readwise's API is returned at most 100 events now.

In this case, should I use the db to develop my own filter strategy? The document says "Use built-in deduping strategies whenever possible (unique, greatest, last) vs developing custom deduping code."

Reply:
I limit the events only to process the first 100 (to prevent the de-duplicate bug if Readwise changes their API to return more at once).

If the limit is changed to 50, or you think I should use db to develop a de-duplicate logic, just tell me.